feat: Stubs are created for referenced declarations in other packages

Masara opened the pull request:
Closes #66

### Summary of Changes

Now limited stubs are created for imported classes from other packages.

lars-reimann commented on tests/safeds_stubgen/stubs_generator/test_generate_stubs.py:
```diff
@@ -68,6 +68,16 @@ def test_file_creation() -> None:
 
 
+def test_file_creation_limited_stubs_outside_package(snapshot_sds_stub: SnapshotAssertion) -> None:
+    # Somehow the stubs get overwritten by other tests, therefore we have to call the function before asserting
```
This does not necessarily have to be done in this PR, though.

Masara commented:
@lars-reimann I changed the stubs-generator the way you said and refactored the tests, are the tests like this all right or should I still try to get the file names with ```glob``` like you mentioned [here](https://github.com/Safe-DS/Stub-Generator/pull/69#discussion_r1509919395)?

Masara commented:
Also, you can merge this PR first.

lars-reimann commented on tests/safeds_stubgen/stubs_generator/test_generate_stubs.py:
```diff
@@ -78,50 +74,29 @@ def test_file_creation_limited_stubs_outside_package(snapshot_sds_stub: Snapshot
+    ]
+)
+def test_stub_creation(file_name: str, snapshot_sds_stub: SnapshotAssertion) -> None:
+    for stub_data in stubs_data:
```
Are you using all entries of `stubs_data`?

Masara commented:
@lars-reimann I changed the tests, now all stubs files should be checked

lars-reimann commented:
Yep, looks great this way and should be easier to extend later on.

lars-reimann commented on src/safeds_stubgen/stubs_generator/_generate_stubs.py:
```diff
@@ -368,13 +456,15 @@ def _create_property_function_string(self, function: Function, indentations: str
-        camel_case_name = self._convert_snake_to_camel_case(name)
+        camel_case_name = name
+        if self.convert_identifiers:
+            camel_case_name = _convert_snake_to_camel_case(name)
```
What happened to this one?

Masara replied:
I renamed the ```_convert_snake_to_camel_case``` function and added the changed you proposed

Masara replied:
Oh nevermind, I misunderstood what you meant, I will add the changes in a moment

lars-reimann replied:
It was mostly about moving the if inside the function, so it's always paired with the call.

Masara replied:
Yes, I fixed that now and removed the if statements

lars-reimann reviewed (approved):
Nice work!

lars-reimann commented:
:tada: This PR is included in version 0.2.0 :tada:

The release is available on:
- `v0.2.0`
- [GitHub release](https://github.com/Safe-DS/Stub-Generator/releases/tag/v0.2.0)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:
